packages/init/test/test-async_hooks.js

```diff
@@ -0,0 +1,79 @@
+/* global globalThis, $262 */
+
+import '../index.js';
+import test from 'ava';
+import { createHook } from 'async_hooks';
+import { setTimeout } from 'timers';
+
+const gcP = (async () => {
+  let gc = globalThis.gc || (typeof $262 !== 'undefined' ? $262.gc : null);
+  if (!gc) {
+    gc = () => {
+      Array.from({ length: 2 ** 24 }, () => Math.random());
+    };
+  }
+  return gc;
+})();
+
+test('async_hooks Promise patch', async t => {
+  const hasSymbols =
+    Symbol.nodeAsyncHooksAsyncId && Symbol.nodeAsyncHooksTriggerAsyncId;
+  let resolve;
+  const q = (() => {
+    const p1 = new Promise(r => (resolve = r));
+    t.deepEqual(
+      Reflect.ownKeys(p1),
+      [],
+      `Promise instances don't start with any own keys`,
+    );
+    harden(p1);
+
+    // The `.then()` fulfillment triggers the "before" hook for `p2`,
+    // which enforces that `p2` is a tracked promise by installing async id symbols
+    const p2 = Promise.resolve().then(() => {});
+    t.deepEqual(
+      Reflect.ownKeys(p2),
+      [],
+      `Promise instances don't start with any own keys`,
+    );
+    harden(p2);
+
+    const testHooks = createHook({
+      init() {},
+      before() {},
+      // after() {},
+      destroy() {},
+    });
+    testHooks.enable();
+
+    // Create a promise with symbols attached
+    const p3 = Promise.resolve();
+    if (hasSymbols) {
+      t.truthy(Reflect.ownKeys(p3));
+    }
```

Comment:
Isn't the array returned by `Reflect.ownKeys` _always_ truthy, even when empty?
```suggestion
    t.is(Reflect.ownKeys(p3).length > 0, hasSymbols);
```

Reply:
Yup, I did a last minute change and missed some things.